Seata 2.0.0 TCC事务发起异常

<!-- Please do not use this issue template to report security vulnerabilities but refer to our [security policy](https://github.com/seata/seata/security/policy). -->

- [x] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description
使用2.0.0版本发起TCC事务时报了如下错误

### Ⅱ. Describe what happened

  在BranchRegister  addBranch 时报错如下错误:

```
java.util.ServiceConfigurationError: com.alipay.sofa.jraft.rpc.RaftRpcFactory: could not find any implementation for class
	at com.alipay.sofa.jraft.util.JRaftServiceLoader.fail(JRaftServiceLoader.java:161)
	at com.alipay.sofa.jraft.util.JRaftServiceLoader.first(JRaftServiceLoader.java:113)
	at com.alipay.sofa.jraft.util.RpcFactoryHelper.<clinit>(RpcFactoryHelper.java:28)
	at com.alipay.sofa.jraft.rpc.ProtobufMsgFactory.<clinit>(ProtobufMsgFactory.java:57)
	at com.alipay.sofa.jraft.rpc.impl.AbstractClientService.<clinit>(AbstractClientService.java:63)
	at com.alipay.sofa.jraft.core.CliServiceImpl.init(CliServiceImpl.java:88)
	at com.alipay.sofa.jraft.core.CliServiceImpl.init(CliServiceImpl.java:73)
	at com.alipay.sofa.jraft.RaftServiceFactory.createAndInitCliService(RaftServiceFactory.java:65)
	at io.seata.server.cluster.raft.RaftServerFactory$SingletonHandler.<clinit>(RaftServerFactory.java:233)
	at io.seata.server.cluster.raft.RaftServerFactory.getInstance(RaftServerFactory.java:78)
	at io.seata.server.session.GlobalSession.addBranch(GlobalSession.java:316)
	at io.seata.server.coordinator.AbstractCore.lambda$branchRegister$0(AbstractCore.java:84)
	at io.seata.server.storage.db.session.DataBaseSessionManager.lockAndExecute(DataBaseSessionManager.java:153)
	at io.seata.server.session.SessionHolder.lockAndExecute(SessionHolder.java:332)
	at io.seata.server.coordinator.AbstractCore.branchRegister(AbstractCore.java:77)
	at io.seata.server.coordinator.DefaultCore.branchRegister(DefaultCore.java:107)
	at io.seata.server.coordinator.DefaultCoordinator.doBranchRegister(DefaultCoordinator.java:299)
	at io.seata.server.AbstractTCInboundHandler$4.execute(AbstractTCInboundHandler.java:184)
	at io.seata.server.AbstractTCInboundHandler$4.execute(AbstractTCInboundHandler.java:179)
	at io.seata.core.exception.AbstractExceptionHandler.exceptionHandleTemplate(AbstractExceptionHandler.java:131)
	at io.seata.server.AbstractTCInboundHandler.handle(AbstractTCInboundHandler.java:179)
	at io.seata.core.protocol.transaction.BranchRegisterRequest.handle(BranchRegisterRequest.java:136)
	at io.seata.server.coordinator.DefaultCoordinator.onRequest(DefaultCoordinator.java:523)
	at io.seata.core.rpc.processor.server.ServerOnRequestProcessor.handleRequestsByMergedWarpMessage(ServerOnRequestProcessor.java:288)
	at io.seata.core.rpc.processor.server.ServerOnRequestProcessor.lambda$onRequestMessage$1(ServerOnRequestProcessor.java:178)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1760)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
02:46:29.513 [ServerHandlerThread_1_5_500] ERROR i.s.c.r.p.s.ServerOnRequestProcessor 191- handle request error: java.lang.NoClassDefFoundError: Could not initialize class io.seata.server.cluster.raft.RaftServerFactory$SingletonHandler
java.util.concurrent.ExecutionException: java.lang.NoClassDefFoundError: Could not initialize class io.seata.server.cluster.raft.RaftServerFactory$SingletonHandler
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2073)
	at io.seata.core.rpc.processor.server.ServerOnRequestProcessor.onRequestMessage(ServerOnRequestProcessor.java:188)
	at io.seata.core.rpc.processor.server.ServerOnRequestProcessor.process(ServerOnRequestProcessor.java:122)
	at io.seata.core.rpc.netty.AbstractNettyRemoting.lambda$processMessage$2(AbstractNettyRemoting.java:280)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.NoClassDefFoundError: Could not initialize class io.seata.server.cluster.raft.RaftServerFactory$SingletonHandler
	at io.seata.server.cluster.raft.RaftServerFactory.getInstance(RaftServerFactory.java:78)
	at io.seata.server.session.GlobalSession.addBranch(GlobalSession.java:316)
	at io.seata.server.coordinator.AbstractCore.lambda$branchRegister$0(AbstractCore.java:84)
	at io.seata.server.storage.db.session.DataBaseSessionManager.lockAndExecute(DataBaseSessionManager.java:153)
	at io.seata.server.session.SessionHolder.lockAndExecute(SessionHolder.java:332)
	at io.seata.server.coordinator.AbstractCore.branchRegister(AbstractCore.java:77)
	at io.seata.server.coordinator.DefaultCore.branchRegister(DefaultCore.java:107)
	at io.seata.server.coordinator.DefaultCoordinator.doBranchRegister(DefaultCoordinator.java:299)
	at io.seata.server.AbstractTCInboundHandler$4.execute(AbstractTCInboundHandler.java:184)
	at io.seata.server.AbstractTCInboundHandler$4.execute(AbstractTCInboundHandler.java:179)
	at io.seata.core.exception.AbstractExceptionHandler.exceptionHandleTemplate(AbstractExceptionHandler.java:131)
	at io.seata.server.AbstractTCInboundHandler.handle(AbstractTCInboundHandler.java:179)
	at io.seata.core.protocol.transaction.BranchRegisterRequest.handle(BranchRegisterRequest.java:136)
	at io.seata.server.coordinator.DefaultCoordinator.onRequest(DefaultCoordinator.java:523)
	at io.seata.core.rpc.processor.server.ServerOnRequestProcessor.handleRequestsByMergedWarpMessage(ServerOnRequestProcessor.java:288)
	at io.seata.core.rpc.processor.server.ServerOnRequestProcessor.lambda$onRequestMessage$1(ServerOnRequestProcessor.java:178)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1760)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: java.lang.ExceptionInInitializerError: Exception java.lang.ExceptionInInitializerError [in thread "ForkJoinPool.commonPool-worker-2"]
	at com.alipay.sofa.jraft.core.CliServiceImpl.init(CliServiceImpl.java:88)
	at com.alipay.sofa.jraft.core.CliServiceImpl.init(CliServiceImpl.java:73)
	at com.alipay.sofa.jraft.RaftServiceFactory.createAndInitCliService(RaftServiceFactory.java:65)
	at io.seata.server.cluster.raft.RaftServerFactory$SingletonHandler.<clinit>(RaftServerFactory.java:233)
	... 23 common frames omitted
02:46:29.513 [ServerHandlerThread_1_4_500] ERROR i.s.c.r.p.s.ServerOnRequestProcessor 191- handle request error: java.lang.ExceptionInInitializerError
java.util.concurrent.ExecutionException: java.lang.ExceptionInInitializerError
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2073)
	at io.seata.core.rpc.processor.server.ServerOnRequestProcessor.onRequestMessage(ServerOnRequestProcessor.java:188)
	at io.seata.core.rpc.processor.server.ServerOnRequestProcessor.process(ServerOnRequestProcessor.java:122)
	at io.seata.core.rpc.netty.AbstractNettyRemoting.lambda$processMessage$2(AbstractNettyRemoting.java:280)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.ExceptionInInitializerError: null
	at com.alipay.sofa.jraft.core.CliServiceImpl.init(CliServiceImpl.java:88)
	at com.alipay.sofa.jraft.core.CliServiceImpl.init(CliServiceImpl.java:73)
	at com.alipay.sofa.jraft.RaftServiceFactory.createAndInitCliService(RaftServiceFactory.java:65)
	at io.seata.server.cluster.raft.RaftServerFactory$SingletonHandler.<clinit>(RaftServerFactory.java:233)
	at io.seata.server.cluster.raft.RaftServerFactory.getInstance(RaftServerFactory.java:78)
	at io.seata.server.session.GlobalSession.addBranch(GlobalSession.java:316)
	at io.seata.server.coordinator.AbstractCore.lambda$branchRegister$0(AbstractCore.java:84)
	at io.seata.server.storage.db.session.DataBaseSessionManager.lockAndExecute(DataBaseSessionManager.java:153)
	at io.seata.server.session.SessionHolder.lockAndExecute(SessionHolder.java:332)
	at io.seata.server.coordinator.AbstractCore.branchRegister(AbstractCore.java:77)
	at io.seata.server.coordinator.DefaultCore.branchRegister(DefaultCore.java:107)
	at io.seata.server.coordinator.DefaultCoordinator.doBranchRegister(DefaultCoordinator.java:299)
	at io.seata.server.AbstractTCInboundHandler$4.execute(AbstractTCInboundHandler.java:184)
	at io.seata.server.AbstractTCInboundHandler$4.execute(AbstractTCInboundHandler.java:179)
	at io.seata.core.exception.AbstractExceptionHandler.exceptionHandleTemplate(AbstractExceptionHandler.java:131)
	at io.seata.server.AbstractTCInboundHandler.handle(AbstractTCInboundHandler.java:179)
	at io.seata.core.protocol.transaction.BranchRegisterRequest.handle(BranchRegisterRequest.java:136)
	at io.seata.server.coordinator.DefaultCoordinator.onRequest(DefaultCoordinator.java:523)
	at io.seata.core.rpc.processor.server.ServerOnRequestProcessor.handleRequestsByMergedWarpMessage(ServerOnRequestProcessor.java:288)
	at io.seata.core.rpc.processor.server.ServerOnRequestProcessor.lambda$onRequestMessage$1(ServerOnRequestProcessor.java:178)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1760)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: java.lang.IllegalStateException: Parse protocol file failed.
	at com.alipay.sofa.jraft.rpc.ProtobufMsgFactory.load(ProtobufMsgFactory.java:88)
	at com.alipay.sofa.jraft.rpc.impl.AbstractClientService.<clinit>(AbstractClientService.java:63)
	... 27 common frames omitted
```

### Ⅲ. Describe what you expected to happen
  能够正常完成事务注册

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

直接发起tcc事务

### Ⅴ. Anything else we need to know?

### Ⅵ. Environment:

- JDK version(e.g. `java -version`):  19 和 21都会出现
- Seata client/server version:  2.0.0
- Database version: mysql 
- OS(e.g. `uname -a`): Alpine Linux 3.18.3

I'm meeting this issue as well, but a little difference
1. I'm using Mac Os 14.1 with M1 cpu.
2. my exception likes below in AT mode
   2023-12-19 21:10:14.575 ERROR --- [ServerHandlerThread_1_49_500] [io.seata.core.rpc.processor.server.ServerOnRequestProcessor] [onRequestMessage] []: handle request error: java.lang.NoClassDefFoundError: Could not initialize class io.seata.server.cluster.raft.RaftServerFactory$SingletonHandler
==>
java.util.concurrent.ExecutionException: java.lang.NoClassDefFoundError: Could not initialize class io.seata.server.cluster.raft.RaftServerFactory$SingletonHandler
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:395) ~[na:na]
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1999) ~[na:na]
	at io.seata.core.rpc.processor.server.ServerOnRequestProcessor.onRequestMessage(ServerOnRequestProcessor.java:188) ~[seata-core-2.0.0.jar:2.0.0]
	at io.seata.core.rpc.processor.server.ServerOnRequestProcessor.process(ServerOnRequestProcessor.java:122) ~[seata-core-2.0.0.jar:2.0.0]
	at io.seata.core.rpc.netty.AbstractNettyRemoting.lambda$processMessage$2(AbstractNettyRemoting.java:280) ~[seata-core-2.0.0.jar:2.0.0]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[na:na]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.100.Final.jar:4.1.100.Final]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]
Caused by: java.lang.NoClassDefFoundError: Could not initialize class io.seata.server.cluster.raft.RaftServerFactory$SingletonHandler
	at io.seata.server.cluster.raft.RaftServerFactory.getInstance(RaftServerFactory.java:78) ~[classes!/:2.0.0]
	at io.seata.server.session.GlobalSession.addBranch(GlobalSession.java:316) ~[classes!/:2.0.0]
	at io.seata.server.coordinator.AbstractCore.lambda$branchRegister$0(AbstractCore.java:84) ~[classes!/:2.0.0]
	at io.seata.server.storage.db.session.DataBaseSessionManager.lockAndExecute(DataBaseSessionManager.java:153) ~[classes!/:2.0.0]
	at io.seata.server.session.SessionHolder.lockAndExecute(SessionHolder.java:332) ~[classes!/:2.0.0]
	at io.seata.server.coordinator.AbstractCore.branchRegister(AbstractCore.java:77) ~[classes!/:2.0.0]
	at io.seata.server.coordinator.DefaultCore.branchRegister(DefaultCore.java:107) ~[classes!/:2.0.0]
	at io.seata.server.coordinator.DefaultCoordinator.doBranchRegister(DefaultCoordinator.java:299) ~[classes!/:2.0.0]
	at io.seata.server.AbstractTCInboundHandler$4.execute(AbstractTCInboundHandler.java:184) ~[classes!/:2.0.0]
	at io.seata.server.AbstractTCInboundHandler$4.execute(AbstractTCInboundHandler.java:179) ~[classes!/:2.0.0]
	at io.seata.core.exception.AbstractExceptionHandler.exceptionHandleTemplate(AbstractExceptionHandler.java:131) ~[seata-core-2.0.0.jar:2.0.0]
	at io.seata.server.AbstractTCInboundHandler.handle(AbstractTCInboundHandler.java:179) ~[classes!/:2.0.0]
	at io.seata.core.protocol.transaction.BranchRegisterRequest.handle(BranchRegisterRequest.java:136) ~[seata-core-2.0.0.jar:2.0.0]
	at io.seata.server.coordinator.DefaultCoordinator.onRequest(DefaultCoordinator.java:523) ~[classes!/:2.0.0]
	at io.seata.core.rpc.processor.server.ServerOnRequestProcessor.handleRequestsByMergedWarpMessage(ServerOnRequestProcessor.java:288) ~[seata-core-2.0.0.jar:2.0.0]
	at io.seata.core.rpc.processor.server.ServerOnRequestProcessor.lambda$onRequestMessage$1(ServerOnRequestProcessor.java:178) ~[seata-core-2.0.0.jar:2.0.0]
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700) ~[na:na]
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1692) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183) ~[na:na]

--- a/changes/en-us/2.x.md
+++ b/changes/en-us/2.x.md
@@ -123,5 +123,6 @@ Thanks to these contributors for their code commits. Please report an unintended
 - [dreamskyvision](https://github.com/dreamskyvision)
 - [jsbxyyx](https://github.com/jsbxyyx)
 - [liuqiufeng](https://github.com/liuqiufeng)
+- [saberyjs](https://github.com/SABERYJS)
 
 Also, we receive many valuable issues, questions and advices from our community. Thanks for you all.

--- a/changes/en-us/2.x.md
+++ b/changes/en-us/2.x.md
@@ -23,6 +23,7 @@ Add changes here for all PR submitted to the 2.x branch.
 - [[#6256](https://github.com/apache/incubator-seata/pull/6256)] fix raft-discovery cannot read registry configuration for seata-all sdk
 - [[#6232](https://github.com/apache/incubator-seata/pull/6232)] convert to utf8mb4 if mysql column is json type
 - [[#6278](https://github.com/apache/incubator-seata/pull/6278)] fix ProtocolV1SerializerTest failed
+- [[#6083](https://github.com/apache/incubator-seata/pull/6324)] fix Parse protocol file failed
 
 
 ### optimize:

--- a/changes/en-us/2.x.md
+++ b/changes/en-us/2.x.md
@@ -23,7 +23,7 @@ Add changes here for all PR submitted to the 2.x branch.
 - [[#6256](https://github.com/apache/incubator-seata/pull/6256)] fix raft-discovery cannot read registry configuration for seata-all sdk
 - [[#6232](https://github.com/apache/incubator-seata/pull/6232)] convert to utf8mb4 if mysql column is json type
 - [[#6278](https://github.com/apache/incubator-seata/pull/6278)] fix ProtocolV1SerializerTest failed
-- [[#6083](https://github.com/apache/incubator-seata/pull/6324)] fix Parse protocol file failed
+- [[#6324](https://github.com/apache/incubator-seata/pull/6324)] fix Parse protocol file failed
 
 
 ### optimize:

--- a/changes/zh-cn/2.x.md
+++ b/changes/zh-cn/2.x.md
@@ -121,5 +121,6 @@
 - [dreamskyvision](https://github.com/dreamskyvision)
 - [jsbxyyx](https://github.com/jsbxyyx)
 - [liuqiufeng](https://github.com/liuqiufeng)
+- [saberyjs](https://github.com/SABERYJS)
 
 同时，我们收到了社区反馈的很多有价值的issue和建议，非常感谢大家。

--- a/changes/zh-cn/2.x.md
+++ b/changes/zh-cn/2.x.md
@@ -23,7 +23,7 @@
 - [[#6256](https://github.com/apache/incubator-seata/pull/6256)] 修复在seata-all sdk下，raft-discovery模块不能读取registry.conf的配置的问题
 - [[#6232](https://github.com/apache/incubator-seata/pull/6232)] 修复在mysql的json类型下出现Cannot create a JSON value from a string with CHARACTER SET 'binary'问题
 - [[#6278](https://github.com/apache/incubator-seata/pull/6278)] 修复 ProtocolV1SerializerTest 失败问题
-- [[#6083](https://github.com/apache/incubator-seata/pull/6324)] 修复 Parse protocol file failed
+- [[#6324](https://github.com/apache/incubator-seata/pull/6324)] 修复 Parse protocol file failed
 
 ### optimize:
 - [[#6031](https://github.com/apache/incubator-seata/pull/6031)] 添加undo_log表的存在性校验

--- a/changes/zh-cn/2.x.md
+++ b/changes/zh-cn/2.x.md
@@ -23,6 +23,7 @@
 - [[#6256](https://github.com/apache/incubator-seata/pull/6256)] 修复在seata-all sdk下，raft-discovery模块不能读取registry.conf的配置的问题
 - [[#6232](https://github.com/apache/incubator-seata/pull/6232)] 修复在mysql的json类型下出现Cannot create a JSON value from a string with CHARACTER SET 'binary'问题
 - [[#6278](https://github.com/apache/incubator-seata/pull/6278)] 修复 ProtocolV1SerializerTest 失败问题
+- [[#6083](https://github.com/apache/incubator-seata/pull/6324)] 修复 Parse protocol file failed
 
 ### optimize:
 - [[#6031](https://github.com/apache/incubator-seata/pull/6031)] 添加undo_log表的存在性校验

--- a/dependencies/pom.xml
+++ b/dependencies/pom.xml
@@ -72,7 +72,7 @@
         <commons-compress.version>1.21</commons-compress.version>
         <ant.version>1.10.12</ant.version>
         <lz4.version>1.7.1</lz4.version>
-        <jraft.version>1.3.13</jraft.version>
+        <jraft.version>1.3.14</jraft.version>
         <netty.version>4.1.86.Final</netty.version>
         <snakeyaml.version>2.0</snakeyaml.version>
         <sofa.hessian.version>4.0.3</sofa.hessian.version>

